src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -0,0 +1,132 @@
+
+  /**
+   * @return a list of the 10 most salient keywords
+   */
+  public static List<String> getKeywords(String key) throws EntityNotFoundException {
+    try {
+      Key datastoreKey = KeyFactory.stringToKey(key);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity result = datastore.get(datastoreKey);
+    
+      Collection<String> keywordCollection = (Collection<String>) result.getProperty("keywords");
+      List<String> keywordList = new ArrayList<>();
+      int index = 0;
+      for (String keyword : keywordCollection) {
+        keywordList.add(keyword);
+    
+        index++;
+        if (index >= MAX_NUM_KEYWORDS) {
+          break;
+        }
+      }
+      return keywordList;
+    } catch (EntityNotFoundException ex) {
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Adds the labels from the analysis of an image into the datastore.
+   * @return the key pointing to the newly added datastore entity.
+   */
+  public static String addKeywords(List<EntityAnnotation> blobAnalysis) throws IOException {
+    String labels = "";
+    for (EntityAnnotation label : blobAnalysis) {
+      labels += label.getDescription() + ", ";
+    }
+    return addToDatastore(labels);
+  }
+
+  /**
+   * Adds the salient keywords from the analysis of a textual user input into the datastore.
+   * @return the key pointing to the newly added datastore entity.
+   */
+  public static String addKeywords(String message) throws IOException {
+    return addToDatastore(message);
+  }
+  
+  /**
+   * Adds salient keywords from the message into the datastore.
+   */
+  private static String addToDatastore(String message) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    AnalyzeEntitiesResponse entityResponse = analyzeEntity(message);
+    List<com.google.cloud.language.v1.Entity> entities = entityResponse.getEntitiesList();
+    TreeSet<com.google.cloud.language.v1.Entity> orderSet = 
+        new TreeSet<>((com.google.cloud.language.v1.Entity o1, com.google.cloud.language.v1.Entity o2) -> {
+      if (o1.getSalience() < o2.getSalience()) {
+        return 1;
+      } else if (o1.getSalience() > o2.getSalience()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
+    for (com.google.cloud.language.v1.Entity entity : entities) {
+      if (entity.getSalience() == 0) {
+        continue;
+      }
+      orderSet.add(entity);
+    }
+    // We must create a new list of Strings as a collection of NLP Entities is not supported by datastore.
+    List<String> keywordList = new ArrayList<>();
+    for (com.google.cloud.language.v1.Entity entity : orderSet) {
+      keywordList.add(entity.getName());
+    }
+    Entity datastoreEntity = new Entity("Keyword");
+    datastoreEntity.setProperty("keywords", keywordList);
+    datastore.put(datastoreEntity);
+    return KeyFactory.keyToString(datastoreEntity.getKey());
+  }
```

Comment:
The block of code from lines 104-111 seem like what you did in the get method, could you make this another method?

Reply:
They're actually quite different. This method adds all keywords into a list to put into data store, whereas the getKeywords method retrieves and puts a cap of 10 keywords into a new list for returning since if there are more than 10 keywords, we don't want more than that.